app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.securemessage.services
+
+import play.twirl.api.Html
+import uk.gov.hmrc.common.message.model.ConversationItem
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import javax.inject.Inject
+import scala.concurrent.Future
+import scala.util.{ Failure, Success }
+import scala.xml.{ Elem, Node, NodeSeq, Unparsed, Utility, Xhtml }
+import scala.util.Try
+import scala.xml.{ SAXParseException, Text }
+import HtmlUtil.*
+import XmlConversion.*
+import uk.gov.hmrc.securemessage.models.RenderType
+
+class HtmlCreatorService @Inject() (servicesConfig: ServicesConfig) {
+
+  def createConversation(
+    latestMessageId: String,
+    messages: List[ConversationItem],
+    replyType: RenderType.ReplyType
+  ): Future[Either[String, Html]] = {
+
+    val conversation = createConversationList(messages.sortWith(_.id > _.id), replyType)
+    val fullConversation = conversation.mkString(Xhtml.toXhtml(<hr/>))
+
+    Future.successful(Right(Html.apply(fullConversation)))
+  }
+
+  private def createConversationList(messages: List[ConversationItem], replyType: RenderType.ReplyType): List[String] =
+    replyType match {
+      case RenderType.CustomerLink =>
+        messages
+          .sortWith(_.id > _.id)
```

Comment:
Looks like this sorting is redundant, you are already doing that before calling this method itself. Also, the code in the 2 cases is almost same expect hasSmallSubject. So, there is a scope of removing duplicated code for better clarity.

Reply:
Ignore my comment if its lift and shift, as you can refactor it later.

Reply:
yes, will refactor it separately. thanks